### PR TITLE
[SPARK-12380] [PYSPARK] use SQLContext.getOrCreate in mllib

### PR DESCRIPTION
**Pull request by davies**
MLlib should use SQLContext.getOrCreate() instead of creating new SQLContext.

### CODE REVIEW

File: python/pyspark/mllib/feature.py [hunk 1]
```diff
@@ -100,8 +100,7 @@ def transform(self, vector):
         :return: normalized vector. If the norm of the input is zero, it
                  will return the input vector.
         """
-        sc = SparkContext._active_spark_context
-        assert sc is not None, "SparkContext should be initialized first"
+        sc = SparkContext.getOrCreate()
```
Comment (jkbradley) [inline python/pyspark/mllib/feature.py]: Actually, sc is not even used here, so it could be removed.

**Comment by jkbradley** [comment]
To confirm: There is no need to use getOrCreate in tests, right?

The other changes look fine.

**Comment by davies** [comment]
@jkbradley For tests, they are fine.

**Comment by jkbradley** [comment]
LGTM pending tests

**Pull request closed without merging**